modules/ossm-migrating-from-smcp-defined-gateways-to-gateway-injection.adoc

```diff
@@ -0,0 +1,142 @@
+      labels: <2>
+        app: istio-ingressgateway
+        istio: ingressgateway
+        sidecar.istio.io/inject: "true"
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto
+      serviceAccountName: istio-ingressgateway
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: istio-ingressgateway
+  namespace: istio-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: secret-reader
+  namespace: istio-system
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get", "watch", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: istio-ingressgateway-secret-reader
+  namespace: istio-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: secret-reader
+subjects:
+  - kind: ServiceAccount
+    name: istio-ingressgateway
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy <3>
+metadata:
+  name: gatewayingress
+  namespace: istio-system
+spec:
+  podSelector:
+    matchLabels:
+      istio: ingressgateway
+  ingress:
+    - {}
+  policyTypes:
+    - Ingress
+---- 
+<1> The gateway injection deployment and all supporting resources should be deployed in the same namespace as the SMCP-defined gateway. 
+<2> Ensure that the labels specified in the pod template include all of the label selectors specified in the Service associated with the existing SMCP-defined gateway.
+<3> Grant access to the new gateway from outside the cluster. This access is required whenever the `spec.security.manageNetworkPolicy` of the `ServiceMeshControlPlane` resource is set to `true`, which is the default setting.
+
+. Verify that the new gateway deployment is successfully handling requests. 
++
+Ensure that the new gateway deployment is successfully handling requests. If access logging was configured in the `ServiceMeshControlPlane` resource view the access logs of the new gateway deployment to confirm the behavior.
+
+. Scale down the old deployment and scale up the new deployment. 
++
+You may gradually shift traffic from the old gateway deployment to the new gateway deployment by running the `oc scale` command at the command line:
++
+[source,terminal]
+----
+$ oc scale -n istio-system deployment/<gateway_deployment> --replicas <new_number_of_replicas>
+----
++
+As you scale, increase the number of replicas for the new gateway deployment and decrease the number of replicas for the old gateway deployment. Continue scaling until the new gateway deployment handles all traffic to the gateway Service. 
+
+. Remove the `app.kubernetes.io/managed-by` label from the gateway Service resource.
++
+Removing the label prevents the Service from being deleted when the gateway is disabled in the `ServiceMeshControlPlane` resource. Run the following command at the CLI:
++
+[source,terminal]
+----
+$ oc label service -n istio-system istio-ingressgateway app.kubernetes.io/managed-by-
+----
```

Comment:
Up to you, but suggestion to move the context of this command to the end of this step:
```suggestion
. Remove the `app.kubernetes.io/managed-by` label from the gateway Service resource by running the following command:
+
[source,terminal]
----
$ oc label service -n istio-system istio-ingressgateway app.kubernetes.io/managed-by-
----
+
Removing the label prevents the Service from being deleted when the gateway is disabled in the `ServiceMeshControlPlane` resource.
```

Reply:
done